var/spack/repos/builtin/packages/cp2k/package.py

```diff
@@ -118,6 +121,12 @@ class Cp2k(MakefilePackage, CMakePackage, CudaPackage, ROCmPackage):
     variant("pytorch", default=False, description="Enable libtorch support")
     variant("quip", default=False, description="Enable quip support")
     variant("mpi_f08", default=False, description="Use MPI F08 module")
+    variant(
+        "unified_memory",
+        default=False,
+        description="Enable unified memory support on Mi250 and Mi300 GPUs",
+        when="+rocm",
```

Comment:
Is this variant also conditional on the specific `amdgpu_target`?

Reply:
good point. For now yes. I will do the necessary mods early next week

Reply:
I guess there's currently no nice way to say
```suggestion
        when="+rocm amdgpu_target=(gfx90a or gfx942)",
```
...? However, you might want to at least add a conflict with all non-gfx90a/gfx942 `amdgpu_targets` for this variant. If CP2K complains loudly enough during configuration I'm not insisting on the conflict, but it'd just be nice to have.

Reply:
@mtaillefumier I've unresolved this comment as I don't see it explicitly handled anywhere. I don't insist on having a conflict for this, but I just wanted to check if you maybe made a change but forgot to push or if you plan to not constrain this?

Reply:
@mtaillefumier sorry, I was unclear: the suggested change is not a valid way of expressing the constraint. I was simply wondering out loud if you or @RMeli might have suggestions for a nicer way of expressing the constraint. If not, no constraint might be easiest.

Reply:
I would impose no constraint as I really doubt that anyone will turn this feature on anything else than Mi250X or Mi300A.